inst/tests/tests.Rraw

```diff
@@ -16067,21 +16067,15 @@ setnames(tbl, latin1, utf8)
 test(2111.04, Encoding(names(tbl)), c('UTF-8', 'UTF-8'))
 
 # fwrite scipen, #2020
-oldSciPen = options('scipen')
-testNo = 2112
-dec = 0L
-for (i in -3:3) {
-  for (j in -3:3) {
-    options(scipen=j)  # influence the value of base::format() being tested against
-    # use format to assure the output is the same as in R
-    v = (-1)^(i+j)
-    test(testNo+(dec<-dec+1L)/100, fwrite(data.table(v)), output=format(v))
-  }
-}
-# non-numeric scipen defaults to 0
-options(scipen = NULL)
-test(testNo+(dec<-dec+1L)/100, fwrite(data.table(10^5)), output='1e+05')
-options(oldSciPen)
+DT = data.table(a=0.0001, b=10^6, c=-20L)
+test(2112.01, fwrite(DT), output=out<-c("a,b,c","1e-04,1e+06,-20"))  # default scipen=0 is ensured in test.data.table()
+test(2112.02, fwrite(DT, scipen=NULL), output=out)
+test(2112.03, fwrite(DT, scipen=999), output=c("a,b,c","0.0001,1000000,-20"))
+test(2112.04, fwrite(DT, scipen=1),   output=c("a,b,c","0.0001,1e+06,-20"))
+test(2112.05, fwrite(DT, scipen=2),   output=c("a,b,c","0.0001,1000000,-20"))
+test(2112.06, fwrite(DT, scipen=-3),  output=c("a,b,c","1e-04,1e+06,-20"))
+# pending ... test(2112.07, fwrite(DT, scipen=-4),  output=c("a,b,c","1e-04,1e+06,-2e+01"))
```

Comment:
any comment why this is disabled would be nice
also scipen=0 test would be god

Reply:
I just found 2112.07 didn't work yesterday and will continue today. scipen=0 is test 2112.01.

Reply:
Now I see what I was going for with the `-1^(i+j)` tests... seems negatives not working:

```
for (i in -10:10) {cat(i, ':', sep = ''); fwrite(DT, scipen = i, col.names = FALSE)}
-10:1e-04,1e+06,-20
-9:1e-04,1e+06,-20
-8:1e-04,1e+06,-20
-7:1e-04,1e+06,-20
-6:1e-04,1e+06,-20
-5:1e-04,1e+06,-20
-4:1e-04,1e+06,-20
-3:1e-04,1e+06,-20
-2:1e-04,1e+06,-20
-1:1e-04,1e+06,-20
0:1e-04,1e+06,-20
1:0.0001,1e+06,-20
2:0.0001,1000000,-20
3:0.0001,1000000,-20
4:0.0001,1000000,-20
5:0.0001,1000000,-20
6:0.0001,1000000,-20
7:0.0001,1000000,-20
8:0.0001,1000000,-20
9:0.0001,1000000,-20
10:0.0001,1000000,-20
```

Reply:
Oh, but `DT$c` is `integer`. Implementation only applied `scipen` to numeric... `write.csv` also ignores `scipen` for `integer`:

```
options(scipen = -4)
write.csv(DT)
"","a","b","c"
"1",1e-04,1e+06,-20
```